infrastructure/k8s/helm-chart/eoapi/values.yaml

```diff
@@ -1,18 +1,88 @@
 # default values for eoapi
 
-gitsha_comment: >
-  To pass `gitSha` in during `helm template` or `helm install` do:
+comment: >
+  some postgres variables are basically secrets that should be passed in via os environment variables.
+  For example, to pass username or password information during `helm template` or `helm install` do:
 
-  $ helm install
-      --set gitSha=$(git rev-parse HEAD | cut -c1-10)
+  $ export GITSHA=$(git rev-parse HEAD | cut -c1-10)
+  $ export PGUSER_VAR=s00pers3cr3t
+  $ export POSTGRES_USER_VAR=s00pers3cr3t
+  $ export POSTGRES_PASSWORD_VAR=superuserfoobar
+  $ export PGPASSWORD_VAR=foobar
+
+  $ helm install \
+      --namespace eoapi \
+      --create-namespace \
+      --set gitSha=$GITSHA \
+      --set db.settings.secrets.PGUSER=$PGUSER \
+      --set db.settings.secrets.POSTGRES_USER=$POSTGRES_USER \
+      --set db.settings.secrets.POSTGRES_PASS=$POSTGRES_PASS \
+      --set db.settings.secrets.PGPASSWORD=$PGPASSWORD \
+      --set db.settings.secrets.POSTGRES_PASSWORD=$POSTGRES_PASSWORD \
+      --set db.settings.secrets.POSTGRES_PASS=$PGPASSWORD \
+      eoapi \
       ./eoapi
 
   $ helm template . \
-      -s template/vector/deployment.yaml \
+      -s template/db/manifest.yaml
       -f values.yaml \
-      --set gitSha=$(git rev-parse HEAD | cut -c1-10)
+      --set gitSha=$GITSHA \
+      --set db.settings.secrets.PGUSER=$PGUSER \
+      --set db.settings.secrets.POSTGRES_USER=$POSTGRES_USER \
+      --set db.settings.secrets.POSTGRES_PASS=$POSTGRES_PASS \
+      --set db.settings.secrets.PGPASSWORD=$PGPASSWORD \
+      --set db.settings.secrets.POSTGRES_PASSWORD=$POSTGRES_PASSWORD \
+      --set db.settings.secrets.POSTGRES_PASS=$PGPASSWORD \
+db:
+  # environment options: "rds" || "k8s"
+  environment: "k8s"
+  enabled: true
+  settings:
+    resources:
+      requests:
+        storage: "100Mi"
+    secrets:
+      POSTGRES_DB: "postgis"
+      POSTGRES_USER: ""
+      POSTGRES_PASSWORD: ""
+      POSTGRES_PORT: "5432"
+      POSTGRES_HOST: "pgstac"
+      POSTGRES_HOST_READER: "pgstac"
+      POSTGRES_HOST_WRITER: "pgstac"
+      DB_MIN_CONN_SIZE: "1"
+      DB_MAX_CONN_SIZE: "10"
+      # default connect: https://www.postgresql.org/docs/current/libpq-envars.html
+      PGDATA: "/var/lib/postgresql/data/pgdata"
+      PGUSER: ""
+      PGPASSWORD: ""
+      PGDATABASE: "postgis"
+      ####################################
+      # STAC requires different names :disappointed:
+      ####################################
```

Comment:
If STAC requires different env vars, can't we still have a single value in our values.yaml but just output it to these different envs within the deployment template? Instead of accepting them as separate `values`?

Reply:
we could but we change the deployment template to use `envFrom` 😆 in this same PR. if we want we can do some acrobatics but I prefer to be explicit

Reply:
Ah, I see - you're also just looping through the settings defined in `values.yaml`.

Yea, so, this is beginning to feel a bit overly abstracted to me - I think I'd prefer the mapping of values from `values` to the `ConfigMap` to be explicit and not just looping through the settings, to give a bit more control in what the interface in `values` is.

I guess the argument is that you are just explicit and always exactly mirroring the variables that each container needs. Which I think is also fair enough as long as we make the `values` are well documented.

So, not a hard opinion, this is probably fine as long as we have clear docs around the values.

Reply:
@batpad : handled this with some acrobatics